an-rope/src/tests.rs

```diff
@@ -12,6 +12,48 @@ fn delete_test_1() {
 }
 
 #[test]
+fn delete_test_2() {
+    let mut r = Rope::new();
+    r.delete((0..0));
+    assert_eq!(&r, "");
+}
+
+#[test]
+fn delete_test_3() {
+    let mut r = Rope::from("this is not fine".to_string());
+    r.delete((..));
+    assert_eq!(&r, "");
+}
+
+#[test]
+fn delete_test_4() {
+    let mut r = Rope::from("this is not fine".to_string());
+    r.delete((11..));
+    assert_eq!(&r, "this is not");
+}
+
+#[test]
+fn delete_test_5() {
+    let mut r = Rope::from("this is not fine".to_string());
+    r.delete((..5));
+    assert_eq!(&r, "is not fine");
+}
+
+#[test]
+#[should_panic(expected = "do not lie on character boundary")]
+fn delete_test_6() {
+    let mut r = Rope::from("this is not fine".to_string());
+    r.delete((..42));
+}
+
+#[test]
+#[should_panic(expected = "attempt to subtract with overflow")]
+fn delete_test_7() {
+    let mut r = Rope::from("this is not fine".to_string());
+    r.delete((12..8)); // lol, fuck you
```

Comment:
or alternatively, if we're gonna leave this unfixed (which is fine w/ me), let's just mention this in the docs under `Panics`?